doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
+
+- PEP8 ignore E501
+- prefer % over format
+- try to avoid generators and decorators
```

Comment:
I think introducing new generators should be discouraged, unless their value is obvious.

Because the intellectual price to understand a new generator is big.